### PR TITLE
tests: Add CLI IDL command tests

### LINKED ISSUE #1784: tests: add tests for the on-chain IDL storage accounts
IDL storage accounts == lang/src/idl.rs.

Tests should test that idl can be uploaded and changed

### PR DESCRIPTION
**Pull request by tomlinton**
Runs a test validator then uses the CLI to init the IDL account and upgrade it via `anchor upgrade` and `anchor write-buffer`/`anchor set-buffer`.

Reading the authority via the CLI required that commitment change in the `anchor init` code.

Closes https://github.com/project-serum/anchor/issues/1784

### CODE REVIEW

File: tests/anchor-cli-idl/programs/idl-commands-one/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+[features]
+no-entrypoint = []
+no-idl = []
+no-log-ix-name = []
+cpi = ["no-entrypoint"]
+default = []
+
+[dependencies]
+anchor-lang = "0.24.2"
```
Comment (paul-schaaf) [inline tests/anchor-cli-idl/programs/idl-commands-one/Cargo.toml:19]: use relative path to `lang`

File: tests/anchor-cli-idl/programs/idl-commands-two/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+[features]
+no-entrypoint = []
+no-idl = []
+no-log-ix-name = []
+cpi = ["no-entrypoint"]
+default = []
+
+[dependencies]
+anchor-lang = "0.24.2"
```
Comment (paul-schaaf) [inline tests/anchor-cli-idl/programs/idl-commands-two/Cargo.toml:19]: relative path

File: tests/anchor-cli-idl/programs/idl-commands-one/Cargo.toml (hunk 1 above)
Comment (tomlinton) [reply tests/anchor-cli-idl/programs/idl-commands-one/Cargo.toml]: thanks!

**Comment by paul-schaaf** [comment]
thanks!

**Pull request merged**